app/models/order_detail.rb

```diff
@@ -129,6 +129,13 @@ def self.for_facility_url(facility_url)
     details
   end
 
+  def self.purchased_active_reservations
+    where(state: ['new', 'inprocess'])
```

Comment:
The combo of `new` and `inprocess` is a recurring theme in this model. I don't know if you want to deal with DRYing them all up now but I want to at least keep it in mind as a TODO.